internal/resource/http.go

```diff
@@ -70,6 +80,103 @@ func (f *Fetcher) UpdateHttpTimeouts(timeouts types.Timeouts) {
 
 	f.client.transport.ResponseHeaderTimeout = time.Duration(responseHeader) * time.Second
 	f.client.client.Transport = f.client.transport
+
+	// Update CAs
+	if len(cas) == 0 {
+		return nil
+	}
+
+	pool, err := x509.SystemCertPool()
+	if err != nil {
+		f.Logger.Err("Unable to read system certificate pool: %s", err)
+		return err
+	}
+
+	for _, ca := range cas {
+		u, err := url.Parse(ca.Source)
+		if err != nil {
```

Comment:
Why not move the url parsing into `getCABlob()` and make that just take the `caReference`?